### PR TITLE
Add NoGradGuard to forward call.

### PR DESCRIPTION
**Pull request by CrossR**
This should help with memory usage (and potentially time...but I doubt that with how quick this is).

Docs for this in pytorch can be found [here](https://docs.pytorch.org/cppdocs/api/typedef_namespacetorch_1abf2c764801b507b6a105664a2406a410.html#_CPPv4N5torch11NoGradGuardE).

As my comment says, this should be equivalent to using `with torch.no_grad():` in Python, which is a pretty standard thing to do, so shouldn't change any actual functionality.

Without the guard:

```
19523 ====================================================================================================
19524 MemoryTracker summary (base-10 MB units used)   
19525 
19526   Peak virtual memory usage (VmPeak)  : 5104.48 MB
19527   Peak resident set size usage (VmHWM): 3312.91 MB
19528   Details saved in: 'mem.db'
19529 ==================================================================================================
```

With the guard:

```
19523 ====================================================================================================
19524 MemoryTracker summary (base-10 MB units used)   
19525 
19526   Peak virtual memory usage (VmPeak)  : 4852.74 MB
19527   Peak resident set size usage (VmHWM): 3082.32 MB
19528   Details saved in: 'mem.db'
19529 ==================================================================================================
```

Left as draft until @cerati can verify the change & @hgreenlee can validate on higher stats than 10 events that this does help memory usage.

### CODE REVIEW

**Comment by knoepfel** [comment]
trigger build

**Comment by tomjunk** [comment]
Approved -- looks like the implementation suggested here:

https://docs.pytorch.org/cppdocs/api/typedef_namespacetorch_1abf2c764801b507b6a105664a2406a410.html#typedef-torch-nogradguard

As long as you do not need the gradients, then we're good.

**Comment by lgarren** [comment]
approve
